modules/sievefilters/modules.php

```diff
@@ -160,6 +160,62 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage sievefilterstoggle/handler
+ */
+class Hm_Handler_sieve_toggle_script_state extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_account', 'script_state', 'sieve_script_name'));
+        if (!$success) {
+            echo json_encode([
+                'status' => 'ERROR',
+                'success' => false,
+                'message' => 'Invalid form submission'
+            ]);
+            return;
+        }
+
+        $imap_account = Hm_IMAP_List::dump($form['imap_account']);
+        $factory = get_sieve_client_factory($this->config);
+
+        try {
+            $client = $factory->init($this->user_config, $imap_account);
+            $state = $form['script_state'] ? 'enabled' : 'disabled';
+            $scripts = $client->listScripts();
+
+            foreach ($scripts as $key => $script) {
+                if ($script == 'main_script') {
+                    $client->removeScripts('main_script');
+                }
+                if ($script == $form['sieve_script_name']) {
+                    if (!$form['script_state']) {
+                        unset($scripts[$key]);
+                    }
+                    $client->renameScript($script, "s{$state}_");
+                }
+            }
+
+            $scripts = $client->listScripts();
+            $main_script = generate_main_script($scripts);
+            save_main_script($client, $main_script, $scripts);
+            $client->activateScript('main_script');
+            $client->close();
+
+            echo json_encode([
```

Comment:
Please never directly echo content in the middle of the script execution - it might have severe side effects. Also, that's a handler module, it is not meant to output data. Instead, use code like $this->out('status', 'OK'); to pass these variables to the output modules. Internally, Cypht detects that it is an ajax request and returns JSON ajax response where you will see res.status == 'OK'.

Reply:
I understand thank you. I correct

Reply:
hello @kroky  I just modified please review

Reply:
Looks good now. Please address the other feedback items as well, so we can merge.